index.js

```diff
@@ -1044,10 +1044,44 @@ function getInnerFields(params, fields) {
 function opContainsAnyField(op, fields) {
   for (var i = 0; i < op.length; i++) {
     var component = op[i];
-    if (component.p.length === 0) {
-      return true;
-    } else if (fields[component.p[0]]) {
+    if (Array.isArray(component.p)) {
+      // json0 op component:
+      //
+      // Each op component has its own array of `p` path segments from doc root.
+      if (component.p.length === 0) {
+        return true;
+      } else if (fields[component.p[0]]) {
+        return true;
+      }
+    } else if (typeof component === 'string') {
+      // json1 field descent from root:
+      //
+      // First string in top-level array means all subsequent operations will be
+      // underneath that top-level field, no need to continue iterating.
+      return fields[component];
+    } else if (hasOwnProperty(component, 'p') ||
```

Comment:
In general, I wonder if we're trying to do too much with heuristics here? It's really hard to determine if an op is a given type, and we may do a bunch of work we don't need to if a certain collection _never_ submits JSON ops (this is the mainline case for our organisation!).

I wonder if this belongs as some sort of helper function plugin exported by this library, which consumers need to enable using ShareDB's [`skipPoll` option](https://github.com/share/sharedb/blob/cc0e3382bf3df4c38c7ccc6bc52da71aff8c9f74/lib/backend.js#L647).

So given a helper function `shouldSkipJson1Poll()`, consumer code might look something like:

```js
const backend = new Backend({
  skipPoll: (collection, id, op, query) => {
    // Let's assume this is easy to determine for a consumer. Each
    // collection probably has a set type that never changes,
    // especially if we're querying!
    if (!isJson1Collection(collection)) return false;
    return shouldSkipJson1Poll(op, query);
  },
});
```

The advantages of this:
 - massively reduces `sharedb-mongo`'s concern leakiness
 - we don't need to use heuristics to determine what types ops are
 - keeps the core of `sharedb-mongo` type-agnostic
 - we can have a "poll by default" fallback, so polling always works correctly
 - 
Disadvantages:
 - effort for consumers
 - obscure optimisation config that many consumers may never find or enable, and pay a performance penalty